### PR TITLE
timeout should propagate to OkHttp

### LINKED ISSUE #2013: testTimeout (RouteResourceClientHCTest) occasionally fails on windows/appveyor
The test should just make sure we consider the timeout setting in the client and we expect the request to fail as 1ms for a connection is usually too low.

But windows either rejects so small timeouts or it is very fast with connecting :)

### PR DESCRIPTION
**Pull request by karussell**
fixes #2013 

Had to move the timeout test into the same package to avoid making the getClientForRequest method public. Now RouteResourceClientHCTest.java has only integration tests going towards localhost.

### CODE REVIEW

File: client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     * Runs the same test with each of the supported traversal modes
+     */
+    @Parameterized.Parameters(name = "POST: {0}, maxUnzippedLength: {1}")
+    public static Collection<Object[]> configs() {
+        return Arrays.asList(new Object[][]{
+                {false, -1},
+                {true, 0},
+                {true, 1000}
+        });
+    }
```
Comment (easbar) [inline client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java:42]: This does not run agains graphhopper.com? do we need the parameterization here?

Comment (karussell) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java:42]: Ok, we would need it for GET and POST but the maxUnzippedLength is indeed misleading and not required.

Comment (karussell) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java:42]: (to test that it is applied to OkHttp for POST and GET)

Comment (easbar) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java:42]: One way to do it with Junit5 would be

```java
@ParameterizedTest(name = "POST={0}")
@BooleanSource(values = {true, false})
public void myTest(boolean usePost) {
}
```

Comment (karussell) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java]: Thanks, will use this as this is perfect here as I have to create it only once :)

Comment (karussell) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java]: Btw: there is no BooleanSource ... it is done via `@ValueSource(booleans = {true, false})` it seems.

Comment (easbar) [reply client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java]: oh sorry my bad

**Pull request merged**